code/src/components/TodoApp.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { tasks } from 'reducers/tasks'
+import { useSelector } from 'react-redux'
+
+import { 
+    FormWrapper, 
+    InputField, 
+    Label, 
+    TextInput, 
+    CheckboxStyled,
+    Submit,
+    ListWrapper,
+    TaskBar,
+    ButtonRow,
+    Cross
+
+} from 'components/StyleComponents'
+
+
+
+export const FormAndList = () => {
+
+    const dispatch = useDispatch()
+
+    const allTasks = useSelector((store) => store.tasks.items)
+
+
+    const [newTask, setTask] = useState()
+
+    const [taskList, setTaskList] = useState(allTasks)
+
+    useEffect(() => { setTaskList(allTasks) }, [allTasks])
+
+
+
+    const taskArray = useSelector((store) => store.tasks.items)
+   
+    const finishedTasks = useSelector((store) => store.tasks.items.filter(t => t.status))
+
+    const unfinishedTasks = useSelector((store) => store.tasks.items.filter(t => !t.status))
+
+
+    const newTaskObject = {
+        id: taskArray.length + 1,
```

Comment:
One convenient way of using id's which will be for sure unique is with a function called uniqid. The use of it is very easy, just installing (npm install uniqid), importing it on imports and then using it here instead of the array length as such: id: uniqid()  😊